src/ImageSharp/IO/ChunkedMemoryStream.cs

```diff
@@ -94,247 +57,133 @@ public override long Position
-            MemoryChunk? chunk = this.memoryChunk;
-            while (chunk != null)
-            {
-                if ((leftUntilAtPos < chunk.Length)
-                        || ((leftUntilAtPos == chunk.Length)
-                        && (chunk.Next is null)))
-                {
-                    // The desired position is in this chunk
-                    this.readChunk = chunk;
-                    this.readOffset = leftUntilAtPos;
-                    break;
-                }
-
-                leftUntilAtPos -= chunk.Length;
-                chunk = chunk.Next;
-            }
-
-            if (this.readChunk is null)
-            {
-                // Position is out of range
-                this.readChunk = backupReadChunk;
-                this.readOffset = backupReadOffset;
-            }
+            this.SetPosition(value);
         }
     }
 
     /// <inheritdoc/>
-    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    public override void Flush()
+    {
+    }
+
+    /// <inheritdoc/>
     public override long Seek(long offset, SeekOrigin origin)
     {
         this.EnsureNotDisposed();
 
-        switch (origin)
+        this.Position = origin switch
         {
-            case SeekOrigin.Begin:
-                this.Position = offset;
-                break;
-
-            case SeekOrigin.Current:
-                this.Position += offset;
-                break;
-
-            case SeekOrigin.End:
-                this.Position = this.Length + offset;
-                break;
-            default:
-                ThrowInvalidSeek();
-                break;
-        }
+            SeekOrigin.Begin => (int)offset,
+            SeekOrigin.Current => (int)(this.Position + offset),
+            SeekOrigin.End => (int)(this.Length + offset),
+            _ => throw new ArgumentOutOfRangeException(nameof(offset)),
+        };
 
-        return this.Position;
+        return this.position;
     }
 
     /// <inheritdoc/>
     public override void SetLength(long value)
         => throw new NotSupportedException();
 
     /// <inheritdoc/>
-    protected override void Dispose(bool disposing)
+    public override int ReadByte()
     {
-        if (this.isDisposed)
-        {
-            return;
-        }
-
-        try
-        {
-            this.isDisposed = true;
-            if (disposing)
-            {
-                ReleaseMemoryChunks(this.memoryChunk);
-            }
-
-            this.memoryChunk = null;
-            this.writeChunk = null;
-            this.readChunk = null;
-            this.chunkCount = 0;
-        }
-        finally
+        this.EnsureNotDisposed();
+        if (this.position >= this.length)
         {
-            base.Dispose(disposing);
+            return -1;
         }
-    }
 
-    /// <inheritdoc/>
-    public override void Flush()
-    {
+        _ = this.Read(this.singleByteBuffer, 0, 1);
+        return MemoryMarshal.GetReference<byte>(this.singleByteBuffer);
```

Comment:
since the `byte[]` path goes via AsSpan(), why not use the span approach directly?

```
Span<byte> buffer = stackalloc byte[1];
return Read(buffer) == 1 ? buffer[0] : -1
```

ideally with `SkipLocalsInit` enabled

(you can't do this for the async path, though)

Reply:
Ah... I'd completely forgotten about that method!

Rather than decorating a public mthod with `[SkipLocalsInit]` i've opted for the following:

```c#
/// <inheritdoc/>
public override int ReadByte()
{
    Unsafe.SkipInit(out byte b);
    return this.Read(MemoryMarshal.CreateSpan(ref b, 1)) == 1 ? b : -1;
}
```

Reply:
Looks much better!